mongorestore/mongorestore.go

```diff
@@ -396,6 +399,14 @@ func (restore *MongoRestore) Restore() Result {
 			}
 			return Result{Err: fmt.Errorf("mongorestore target '%v' invalid: %v", restore.TargetDirectory, err)}
 		}
+		preludeFileExists, err := restore.ReadPreludeMetadata(target)
+		if !preludeFileExists {
+			// don't error out here because older dump versions will not prelude.json
+			log.Logvf(log.DebugLow, "no prelude metadata found in target directory or parent, skipping")
+		} else if err != nil {
+			return Result{Err: fmt.Errorf("error reading dump metadata from prelude.json: %v", err)}
```

Comment:
```suggestion
			return Result{Err: fmt.Errorf("error reading dump metadata from prelude.json: %w", err)}
```

Reply:
It’s probably better not to name the file specifically since it can optionally be `.json.gz` … ?